modules/mobile/app/helpers/mobile/list_filter.rb

```diff
@@ -7,88 +7,84 @@ class ListFilter
 
-    def initialize(collection, filter_params)
-      @collection = collection
+    def initialize(list, filter_params)
+      @list = list
       @filter_params = filter_params
     end
 
     # Accepts params:
-    #   @collection - a Common::Collection of Common::Base models
+    #   @list - an array of Common::Resource or Common::Base models
     #   @filter_params - should be an ActionController::Parameters object which should be passed in from the
     #     controller via @params[:filter]. This will pass in another ActionController::Parameters object.
-    # Returns: a new Common::Collection of Common::Base models that match the provided filters
-    def self.matches(collection, filter_params)
-      filterer = new(collection, filter_params)
+    # Returns: an array of Common::Resource or Common::Base models that match the provided filters
+    def self.matches(list, filter_params)
+      filterer = new(list, filter_params)
       filterer.result
     end
 
     def result
       validate!
-      metadata = @collection.metadata.merge(filter: filters)
-      Common::Collection.new(data: matches, metadata: metadata, errors: @collection.errors)
+      [matches, nil]
     rescue FilterError => e
-      @collection.errors[:filter_error] = e.message if valid_collection?
       log_exception_to_sentry(e, extra_context_for_errors)
-      @collection
+      [@list, { filter_error: e.message }]
     rescue => e
-      @collection.errors[:filter_error] = 'unknown filter error'
       log_exception_to_sentry(e, extra_context_for_errors)
-      @collection
+      [@list, { filter_error: 'unknown filter error' }]
     end
 
-    # not adding full collection to extra context because it could be a large amount of data,
-    # could expose PII, and isn't likely to be relevant
     def extra_context_for_errors
       extra_context = {}
       extra_context[:filters] = filters if filter_is_parameters?
-      extra_context[:collection_models] = filterable_models.map(&:to_s) if valid_collection?
+      extra_context[:list_models] = filterable_models.map(&:to_s) if valid_list?
       extra_context
     end
 
     private
 
     def matches
-      @collection.data.select { |record| record_matches_filters?(record) }
+      @list.select { |record| record_matches_filters?(record) }
     end
 
     def record_matches_filters?(record)
       filters.all? do |match_attribute, operations_and_values|
         match_attribute = match_attribute.to_sym
-        model_attribute = model_attributes.find { |att| att.name == match_attribute }
-        coercer = model_attribute.coercer
 
         operations_and_values.each_pair.all? do |operation, value|
-          coerced_value = coercer.call(value)
-
           case operation.to_sym
           when :eq
-            record[match_attribute] == coerced_value
-          when :notEq
-            record[match_attribute] != coerced_value
+            record[match_attribute].to_s == value
```

Comment:
I was a little sad to move away from the coercer, but Common::Resource is much more stripped down that Common::Base and accessing its coercers was not as easy. I believe that simply converting the model attribute to a string should be operationally safe and achieve the same result.